commit 55f3b5767f42a38f9d3e2cd6161577fe8f1ef141
Author: Luis Lozano <llozano@google.com>
Date:   Thu May 21 11:28:54 2020 -0700

    Disable "[CodeGen][ARM] Error when writing to specific reserved registers in inline asm"

    This is a logical revert of d85b3877dcd2.

    Will do this temporarily until we get a solution for the bug.

    BUG=chromium:1084372
    TEST=CQ testing.

diff --git a/llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp b/llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp
index 51539f38188..f9f85d7a812 100644
--- a/llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp
+++ b/llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp
@@ -8146,8 +8146,9 @@ void SelectionDAGBuilder::visitInlineAsm(const CallBase &Call) {
       const MachineFunction &MF = DAG.getMachineFunction();
       const TargetRegisterInfo &TRI = *MF.getSubtarget().getRegisterInfo();
       for (unsigned Reg : OpInfo.AssignedRegs.Regs) {
+        // Disable this error for now. See crosbug/1084372
         if (Register::isPhysicalRegister(Reg) &&
-            TRI.isInlineAsmReadOnlyReg(MF, Reg)) {
+            TRI.isInlineAsmReadOnlyReg(MF, Reg) && false) {
           const char *RegName = TRI.getName(Reg);
           emitInlineAsmError(Call, "write to reserved register '" +
                                        Twine(RegName) + "'");
